ampd/src/json_rpc.rs

```diff
@@ -16,14 +19,20 @@ where
     P: JsonRpcClient,
 {
     provider: P,
+    monitoring_client: monitoring::Client,
+    handler_label: String,
```

Comment:
the json client shouldn't care about the handler. Given that the chain name looks like the important corresponding factor across multiple metrics, use `chain: ChainName` instead

Reply:
the rpc is the same across different handlers for the same chain anyway